Add a message into the order to show who is the employee who created the order

Prestaworks opened the pull request:
Fix for issue 9676

<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/8/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | In order to fix #9676 I add a message into the order to show who is the employee who created the order
| Type?             | bug fix
| Category?         |  BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #9676. and also #9783 
| Related PRs       | 
| How to test?      | Add new order from BO, employee information is added as message on order.
| Possible impacts? |


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

Prestaworks committed:
issue_9676 and issue 9783

Show messages from both ps_customer_messages and ps_messages on the order view.

Progi1984 commented on src/Adapter/Customer/CustomerDataProvider.php:
```diff
@@ -114,15 +115,29 @@ public function getCustomerMessages(int $customerId, ?int $orderId = null, ?int
+            }
+        }
+
+        if (is_array($customerMessages)) {
+            $count = count($customerMessages);
+        } else {
+            $count = 0;
+        }
```
```suggestion
        $count = is_array($customerMessages) ? count($customerMessages) : 0;
```

Progi1984 reviewed (commented):
A small feedback

Hlavtox commented:
Hi @Prestaworks, before merging this, I think it could be good to explore the possibilities of clearing the situation explained by @matks here:
https://github.com/PrestaShop/PrestaShop/issues/9783#issuecomment-827477060

- Unify the table where all messages will be stored. From both FO checkout, FO order tracking, FO contact form, BO order create etc. So we don't have the data split like this.
- Migrate data with upgrade script.
- Deprecate old functions and modify them to store data into the correct table.

Prestaworks commented:
> Hi @Prestaworks, before merging this, I think it could be good to explore the possibilities of clearing the situation explained by @matks here: [#9783 (comment)](https://github.com/PrestaShop/PrestaShop/issues/9783#issuecomment-827477060)
> 
>     * Unify the table where all messages will be stored. From both FO checkout, FO order tracking, FO contact form, BO order create etc. So we don't have the data split like this.
> 
>     * Migrate data with upgrade script.
> 
>     * Deprecate old functions and modify them to store data into the correct table.

customer message and order message are a bit different, the order message is supposed to log warnings, transaction numbers and such so I am not sure it's a good idea to just remove ps_message table and store everything as a customer message. But if that's something the maintainers think is a good solution, it should not be a big issue to fix.

However, there are a few other hickups that needs to be fixed also, the automatic tests expects 0 messages, even for orders created from the backend and since this PR will get the message for "create by", the test will fail, so the test has to be rewritten also to only get public messages.

matks commented:
(funny thing: I check github notifications by email and both of you are displayed as "Daniel" so I see Daniel talking to Daniel)

Hlavtox commented:
> customer message and order message are a bit different, the order message is supposed to log warnings, transaction numbers

Definitely not my friend. Both field should contain messages from employee or customer, not any system data. Transaction numbers belong to custom table for the payment module or ps_payment table. Logs belong to custom table for the payment module or to ps_logs table.

Prestaworks commented:
> 

I am not saying how I would like it to be, I am just saying how it has been used in the past.

this for instance is a bit of legacy in paymentModule class
```
if (!$secure_key) {
                $message .= '<br />' . $this->trans('Warning: the secure key is empty, check your payment account before validation', [], 'Admin.Payment.Notification');
            }
```
This is then saved in ps_message, never to be displayed anywhere :)

Maybe, we could start by making sure ps_message is displayed properly and that the tests work? 
And then, as a new feature in a future release we could remove ps_message? removing it would take some effort on the upgrade module also so that the data is moved before the table is dropped, and that module seems to be really difficult to make changes for.

Prestaworks committed:
issue 9676

PaymentModule, remove duplicate storage of order message.
CustomerDataProvider back to original code
GetOrderForViewingHandler get two lists of messages into one list for display.
OrderMessageContext change test to check if no public messages are displayed.

Prestaworks committed:
Update GetOrderForViewingHandler.php

Prestaworks committed:
Update GetOrderForViewingHandler.php

Prestaworks committed:
Update CustomerDataProvider.php

matthieu-rolland reviewed (commented):
Do we have a green light from @PrestaShop/product-team for this feature that apparently disappeared with ps 1.7.7 ?

Commit:
fix mockups integration

atomiix commented on classes/PaymentModule.php:
```diff
@@ -519,8 +519,6 @@ public function validateOrder(
             $old_message = Message::getMessageByCartId((int) $this->context->cart->id);
             if ($old_message && !$old_message['private']) {
                 $update_message = new Message((int) $old_message['id_message']);
```
I guess this is useless now? You can use `$old_message->message` to populate `$customer_thread->message`

atomiix reviewed (commented):
I see that you're displaying both OrderMessage and CustomerThread here but I can't see where the actual message that says who created the order is added?

Prestaworks commented:
> e

I think that was default behaviour, it was just not displayed on the order view.
